### DISCUSSION TITLE
Is ReadingExceptionOccurred working?

### DISCUSSION THREAD

**Discussion by dprice** [JoshClose/CsvHelper#1763] [Q&A]
I upgraded to the latest version (26.1.0) from an incredibly early version (12.1.2) and cannot seem to get `ReadingExceptionOccurred` to fire. I even copied the unit tests from the `DetectColumnCountChangesTests.cs` file:
https://github.com/JoshClose/CsvHelper/blob/0d753ff09294b425e4bc5ab346145702eeeb1b6f/tests/CsvHelper.Tests/Reading/DetectColumnCountChangesTests.cs

and all tests passed except for `WillThrowOnMissingFieldStillWorksTest`.

Does anyone have this working?

**Comment by dprice** [comment]
The test seem incorrect since the exception type for `ReadingExceptionOccurred` is `CsvHelperException` and `MissingFieldException` is not derived from `CsvHelperException` so this test will never pass.

**Reply by JoshClose** [reply]
`MissingFieldException -> ReaderException -> CsvHelperException`

That test passes for me.
